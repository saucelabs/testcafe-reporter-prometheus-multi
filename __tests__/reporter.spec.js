const createReporter = require('./../src/');

describe('testcafe-reporter-prometheus-multi', () => {
    const fixture = {
        name: 'My set of tests',
        path: '/some/path',
        meta: {
            app: 'my-app',
            owner: 'me',
        },
    };

    const test1 = {
        name: 'Test Number One',
        testRunInfo: {
            errs: [],
            skipped: false,
        },
        meta: {},
    };

    const test2 = {
        name: 'Test Number Two',
        testRunInfo: {
            errs: [new Error()],
            skipped: false,
        },
        meta: {},
    };

    const test3 = {
        name: 'Test Number Three',
        testRunInfo: {
            errs: [],
            skipped: true,
        },
        meta: {},
    };
    const test4 = {
        name: 'Test Number Four',
        testRunInfo: {
            errs: [],
            skipped: false,
        },
        meta: {
            app: 'someone-elses-app',
            owner: 'someone-else',
        },
    };
    const test5 = {
        name: 'Test Number Five',
        testRunInfo: {
            errs: [new Error()],
            skipped: false,
        },
        meta: {
            owner: 'someone-else',
        },
    };

    let reporter;

    beforeEach(() => {
        reporter = createReporter();
        reporter.write = jest.fn();
    });

    it('reports three tests, one of each status', () => {
        reporter.reportTaskStart();
        reporter.reportFixtureStart(fixture.name, fixture.path, fixture.meta);
        [test1, test2, test3].forEach((test) => (
            reporter.reportTestDone(test.name, test.testRunInfo, test.meta)
        ));
        reporter.reportTaskDone();
        expect(reporter.write).toMatchSnapshot();
    });

    it('test meta properties app & owner overwrite fixture properties', () => {
        reporter.reportTaskStart();
        reporter.reportFixtureStart(fixture.name, fixture.path, fixture.meta);
        [test1, test2, test4, test5].forEach((test) => (
            reporter.reportTestDone(test.name, test.testRunInfo, test.meta)
        ));
        reporter.reportTaskDone();
        expect(reporter.write).toMatchSnapshot();
    });
});
