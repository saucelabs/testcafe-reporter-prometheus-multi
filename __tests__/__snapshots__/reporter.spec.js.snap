// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testcafe-reporter-prometheus-multi reports three tests, one of each status 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "# HELP e2e_passed_tests Passed tests
# TYPE e2e_passed_tests gauge
e2e_passed_tests{app=\\"my-app\\",owner=\\"me\\"} 1

# HELP e2e_skipped_tests Skipped tests
# TYPE e2e_skipped_tests gauge
e2e_skipped_tests{app=\\"my-app\\",owner=\\"me\\"} 1

# HELP e2e_failed_tests Failed tests
# TYPE e2e_failed_tests gauge
e2e_failed_tests{app=\\"my-app\\",owner=\\"me\\"} 1
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`testcafe-reporter-prometheus-multi test meta properties app & owner overwrite fixture properties 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "# HELP e2e_passed_tests Passed tests
# TYPE e2e_passed_tests gauge
e2e_passed_tests{app=\\"my-app\\",owner=\\"me\\"} 1
e2e_passed_tests{app=\\"someone-elses-app\\",owner=\\"someone-else\\"} 1
e2e_passed_tests{app=\\"my-app\\",owner=\\"someone-else\\"} 0

# HELP e2e_skipped_tests Skipped tests
# TYPE e2e_skipped_tests gauge
e2e_skipped_tests{app=\\"my-app\\",owner=\\"me\\"} 0
e2e_skipped_tests{app=\\"someone-elses-app\\",owner=\\"someone-else\\"} 0
e2e_skipped_tests{app=\\"my-app\\",owner=\\"someone-else\\"} 0

# HELP e2e_failed_tests Failed tests
# TYPE e2e_failed_tests gauge
e2e_failed_tests{app=\\"my-app\\",owner=\\"me\\"} 1
e2e_failed_tests{app=\\"someone-elses-app\\",owner=\\"someone-else\\"} 0
e2e_failed_tests{app=\\"my-app\\",owner=\\"someone-else\\"} 1
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
